### DISCUSSION TITLE
How to use non ASCII characters as password

### DISCUSSION THREAD

**Discussion by pkuehnel** [dotnet/MQTTnet#1523] [Q&A] [answered]
I am using MQTTnet as a client and am trying to implement password authentication. As I am using `Encoding.ASCII` I am not able to use any non ASCII chars as password. How can I resolve this?

My current code:

```
var ascii = Encoding.ASCII;
var passwordBytes = ascii.GetBytes("MyPassword");
mqttClientOptions.Credentials = new MqttClientCredentials("MyUser", passwordBytes);
```

**Comment by chkr1011** [comment, marked as answer]
Try using Encoding.UTF8 instead.

**Reply by pkuehnel** [reply]
Hmm. I tried Unicode which did not work (Could have been me doing something wrong). `Encoding.UTF8` is woking, thank you.

**Reply by SeppPenner** [reply]
@pkuehnel This strongly depends on what you want to do... MQTTnet doesn't do anything with a byte payload in specific. This means, if you encode your data in Unicode, you should be able to decode it in Unicode on retrieval in the server again (The same for ASCII and UTF8 and the probably 50 other encodings that are out there).
